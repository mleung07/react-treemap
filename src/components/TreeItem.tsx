import styled from "styled-components";
import { Item } from "../features/data";

interface Props extends Item {
  weightPerRow: number;
  rows: number;
}

const Box = styled.div<Props>`
  display: inline-block;
  border: 1px solid black;
  /* calculate height by dividing no. of rows in % */
  height: ${(props) => `calc(100% / ${props.rows})`};
  /* calculate width by dividing weight by weight per row in % */
  width: ${(props) => `calc(100% / ${props.weightPerRow} * ${props.weight})`};
  background-color: ${(props) => {
    switch (Math.sign(props.value)) {
      case 1:
        return "green";
      case -1:
        return "red";
      default:
        return "grey";
    }
  }};
`;

const Wrapper = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

const TreeItem = (props: Props) => {
  return (
    <Box {...props}>
      <Wrapper>
        <div>{props.name}</div>
        <div>{props.value * 100}%</div>
      </Wrapper>
    </Box>
  );
};

export default TreeItem;
