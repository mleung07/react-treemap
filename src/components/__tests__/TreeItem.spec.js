import React from "react";
import { render } from "@testing-library/react";
import TreeItem from "../TreeItem";

test("renders positive valued TreeItem component", () => {
  const testProps = {
    rows: 3,
    weightPerRow: 5,
    weight: 4,
    name: "+ve",
    value: 0.05,
  };
  const elem = render(<TreeItem {...testProps} />);

  expect(elem).toMatchSnapshot();
});

test("renders negative valued TreeItem component", () => {
  const testProps = {
    rows: 3,
    weightPerRow: 5,
    weight: 4,
    name: "-ve",
    value: -0.05,
  };
  const elem = render(<TreeItem {...testProps} />);

  expect(elem).toMatchSnapshot();
});

test("renders zero valued TreeItem component", () => {
  const testProps = {
    rows: 3,
    weightPerRow: 5,
    weight: 4,
    name: "zero",
    value: 0,
  };
  const elem = render(<TreeItem {...testProps} />);

  expect(elem).toMatchSnapshot();
});
