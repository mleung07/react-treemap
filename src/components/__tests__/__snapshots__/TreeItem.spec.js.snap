// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders negative valued TreeItem component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBQB eByzUK"
        name="-ve"
        rows="3"
        value="-0.05"
      >
        <div
          class="sc-gsTEea gpYexz"
        >
          <div>
            -ve
          </div>
          <div>
            -5
            %
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBQB eByzUK"
      name="-ve"
      rows="3"
      value="-0.05"
    >
      <div
        class="sc-gsTEea gpYexz"
      >
        <div>
          -ve
        </div>
        <div>
          -5
          %
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders positive valued TreeItem component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBQB gMHsFy"
        name="+ve"
        rows="3"
        value="0.05"
      >
        <div
          class="sc-gsTEea gpYexz"
        >
          <div>
            +ve
          </div>
          <div>
            5
            %
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBQB gMHsFy"
      name="+ve"
      rows="3"
      value="0.05"
    >
      <div
        class="sc-gsTEea gpYexz"
      >
        <div>
          +ve
        </div>
        <div>
          5
          %
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders zero valued TreeItem component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBQB hdkens"
        name="zero"
        rows="3"
        value="0"
      >
        <div
          class="sc-gsTEea gpYexz"
        >
          <div>
            zero
          </div>
          <div>
            0
            %
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBQB hdkens"
      name="zero"
      rows="3"
      value="0"
    >
      <div
        class="sc-gsTEea gpYexz"
      >
        <div>
          zero
        </div>
        <div>
          0
          %
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
